### ISSUE TITLE
Properties may be defined without a field causing a NullPointerException

### ISSUE THREAD

**Issue by langecode** [Nykredit/jackson-dataformat-hal#3]
Jackson will consider properties which are only defined through a getter which causes the deserialization to throw a `NullPointerException`.

Consider the following POJO.

```
class TestClass {
    @JsonProperty
    private long lastModified;

    public long getTime() {
        return lastModified;
    }
}
```

`time` is treated as a property but has no directly associated field. Causing a NullPointerException in line 39 of `ReservedProperty`.

**Comment by langecode** [comment]
One workaround is to rename the field to match the getter. I.e., the below would work.
```
class TestClass {
    @JsonProperty("lastModified")
    private long time;

    public long getTime() {
        return time;
    }
}
```

**Comment by langecode** [comment]
This is actually solved as part of #5
